mrp_subcontract_location/models/purchase_order.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#  licence AGPL version 3 or later
+#  see licence in __openerp__.py or http://www.gnu.org/licenses/agpl-3.0.txt
+#  Copyright (C) 2015 Akretion (http://www.akretion.com).
+#
+##############################################################################
+from odoo import models, fields, api, exceptions, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    warehouse_id = fields.Many2one(related='picking_type_id.warehouse_id',
+        string='Warehouse',
+        help="Technical field used to display the Drop Ship Address",
+        readonly=True)
+
+    def button_approve(self):
+        res = super(PurchaseOrder, self).button_approve()
+        for purchase in self:
+            for line in purchase.order_line:
+                if line._is_service_procurement():
+                    mo = line.procurement_ids and \
```

Comment:
no need to test on line.procurement_ids if it's a service_procurement()

Reply:
le _is_service_procurement vérifie juste que le produit est de type service et avec la case subcontracted service cochée, mais si on ajoute un produit de ce genre manuellement dans une PO, elle aura ni procurement, ni MO de liée ?

Reply:
is_service_**procurement**__  le nom implique un procurement donc faudrait faire ce check dans cette méthode.

(d'ailleurs on devrait peut être renommer en is_procurement_service),

Reply:
https://github.com/akretion/ak-odoo-incubator/pull/87/commits/f7b21c5d81667594aa529fc594ced7d9bb3a4610